graylog2-web-interface/src/components/common/EntityDataTable/hooks/useCalculateStaticColumnWidths.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import { useMemo } from 'react';
+import type { DefaultTheme } from 'styled-components';
+import { useTheme } from 'styled-components';
+
+import { SORT_ICON_WIDTH } from 'components/common/EntityDataTable/SortIcon';
+import type { ColumnRenderersByAttribute, EntityBase } from 'components/common/EntityDataTable/types';
+import type { ColumnSchema } from 'components/common/EntityDataTable';
+import { DRAG_HANDLE_WIDTH } from 'components/common/SortableList/DragHandle';
+import useMeasureTextWidth from 'components/common/EntityDataTable/hooks/useMeasureTextWidth';
+
+const HEADER_PADDING = 10; // px
+const DEFAULT_FONT_WEIGHT = 700;
```

Comment:
For example the header padding is currently defined in bootstrap styles, this is something we can potentially improve by defining the padding explicitly in the `Table` component and importing it here.